python/subunit/v2.py

```diff
@@ -72,13 +72,32 @@ def has_nul(buffer_or_bytes):
         return NUL_ELEMENT in buffer_or_bytes
 
 
+def read_exactly(stream, size):
+    """Read exactly size bytes from stream.
+
+    :param stream: A file like object to read bytes from. Must support
+        read(<count>) and return bytes.
+    :param size: The number of bytes to retrieve.
+    """
+    data = b''
+    remaining = size
+    while remaining:
+        read = stream.read(remaining)
+        if not len(read):
```

Comment:
I initially wrote "So for `read` None and `b''` have different semantics - one shouldn't ever use a truthy test for it unless the file descriptors mode is completely controlled (and we don't control the mode of a file passed in here)."

Then I realised that that is on `len(read)` which is integral and thus avoiding the type issue (if read is None the expression will throw with length of an unsized type) - I think I'd prefer

`if len(read) == 0:` even though the truthy test is valid, just for easier reasoning.

Reply:
Sure, done.